src/printer.js

```diff
@@ -78,12 +78,26 @@ function genericPrint(path, options, printPath, args) {
     // responsible for printing node.decorators.
     !util.getParentExportDeclaration(path)
   ) {
-    const separator = node.decorators.length === 1 &&
-      (node.decorators[0].expression.type === "Identifier" ||
-        node.decorators[0].expression.type === "MemberExpression")
-      ? " "
-      : hardline;
+    let shouldBeInlined = false;
+    if (node.decorators.length === 1) {
+      let decorator = node.decorators[0];
```

Comment:
For some reason,

```js
class Person {
  @inject leftLeg: Leg;
}
```

is not wrapped in a Decorator node, which is different than Babylon which does have one. So I need to do this check here to account for both options.